### PR TITLE
mypyify

### PR DESCRIPTION
**Pull request by stoprightthere**

### CODE REVIEW

File: geometric_kernels/spaces/eigenfunctions.py [hunk 1]
```diff
@@ -119,6 +119,11 @@ def dim_of_eigenspaces(self) -> B.Numeric:
         """Dimension of eigenspaces. For generic Eigenfunctions, it is an array of ones. For Eigenfunctions with addition theorem, it is the number of eigenfunctions in each level."""
         return B.ones(self.num_eigenfunctions)
 
+    @abc.abstractproperty
+    def num_eigenfunctions_per_level(self) -> B.Numeric:
```
Comment (vabor112) [inline geometric_kernels/spaces/eigenfunctions.py:123]: TBH I don't understand why this works. For hyperspheres, for example, this function returns a list, and I believe that list is not an instance of `B.Numeric`.
image

File: geometric_kernels/spaces/homogeneous_spaces.py [hunk 2]
```diff
@@ -148,15 +148,15 @@ def num_levels(self) -> int:
     @property
     def num_eigenfunctions(self) -> int:
         """Number of eigenfunctions, M"""
-        return self._num_eigenfunctions
+        return self._num_levels
```
Comment (vabor112) [inline geometric_kernels/spaces/homogeneous_spaces.py:151]: Cannot be true, levels are characters for the homogeneous spaces, there's a bunch of eigenfunctions per each level.

**Review by vabor112 — COMMENTED** [review]
Thanks a lot for doing this! Pls take a look at a couple of minor comments I left.

File: geometric_kernels/spaces/homogeneous_spaces.py (hunk 2 above)
Comment (stoprightthere) [reply geometric_kernels/spaces/homogeneous_spaces.py:151]: Then why does `num_eigenfunctions_per_level` return a list of ones?

File: geometric_kernels/spaces/eigenfunctions.py (hunk 1 above)
Comment (stoprightthere) [reply geometric_kernels/spaces/eigenfunctions.py:123]: Interesting! Maybe we need a stricter mypy config

File: geometric_kernels/spaces/homogeneous_spaces.py (hunk 2 above)
Comment (stoprightthere) [reply geometric_kernels/spaces/homogeneous_spaces.py:151]: This seems to follow the convention as in the rest of the code, that ties in the methods `__call__`, `num_eigenfunctions`, `num_levels`, `num_eigenfunctions_per_level`, and `get_repeated_eigenvalues`. Seems like we can safely make this un-implemented?

Comment (vabor112) [reply geometric_kernels/spaces/homogeneous_spaces.py:151]: Okay. It would be great if you could create this other PR soon, before we forget about this.

**Pull request merged**